pkg/cache/internal/informers.go

```diff
@@ -558,3 +592,50 @@ func restrictNamespaceBySelector(namespaceOpt string, s Selector) string {
 	}
 	return ""
 }
+
+// eitherChan returns a channel that is closed when either of the input channels are signaled.
+// The caller must call the returned CancelFunc to ensure no resources are leaked.
+func eitherChan(a, b <-chan struct{}) (<-chan struct{}, context.CancelFunc) {
```

Comment:
These utilities might be useful elsewhere as well, could we:
- Move them under a new package like `pkg/internal/syncs`
- Use generics so we can reuse for other types of channels as well
- Add some tests

Reply:
Absolutely!